iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  OnboardingScreenView.swift
+//  iosApp
+//
+//  Created by Horn, Melody on 2024-10-25.
+//  Copyright © 2024 MBTA. All rights reserved.
+//
+
+import CoreLocation
+import shared
+import SwiftUI
+
+struct OnboardingScreenView: View {
+    let screen: OnboardingScreen
+    let advance: () -> Void
+
+    let createLocationFetcher: () -> any LocationFetcher
+    let onboardingRepository: IOnboardingRepository
+    let settingUseCase: SettingUsecase
+    @State private var locationFetcher: LocationFetcher?
+    private let locationPermissionHandler: LocationPermissionHandler
+
+    let inspection = Inspection<Self>()
+
+    init(
+        screen: OnboardingScreen,
+        advance: @escaping () -> Void,
+        createLocationFetcher: @escaping () -> any LocationFetcher = { CLLocationManager() },
+        onboardingRepository: IOnboardingRepository = RepositoryDI().onboarding,
+        settingUseCase: SettingUsecase = UsecaseDI().settingUsecase
+    ) {
+        self.screen = screen
+        self.advance = advance
+        self.createLocationFetcher = createLocationFetcher
+        self.onboardingRepository = onboardingRepository
+        self.settingUseCase = settingUseCase
+        locationPermissionHandler = LocationPermissionHandler(
+            screen: screen,
+            onboardingRepository: onboardingRepository,
+            advance: advance
+        )
+    }
+
+    var body: some View {
+        VStack {
+            switch screen {
+            case .feedback:
+                Text(
+                    "MBTA Go is just getting started! We’re actively making improvements based on feedback from riders like you."
+                )
+                Button("Get started") {
+                    Task {
+                        try? await onboardingRepository.markOnboardingCompleted(screen: screen)
+                        advance()
+                    }
+                }
+            case .hideMaps:
+                Text("For VoiceOver users, we’ll keep maps hidden by default unless you tell us otherwise.")
+                Button("Hide maps") {
+                    Task {
+                        try await settingUseCase.set(setting: .hideMaps, value: true)
+                        try? await onboardingRepository.markOnboardingCompleted(screen: screen)
+                        advance()
+                    }
+                }
+                Button("Show maps") {
+                    Task {
+                        try await settingUseCase.set(setting: .hideMaps, value: false)
+                        try? await onboardingRepository.markOnboardingCompleted(screen: screen)
+                        advance()
+                    }
+                }
+            case .location:
```

Comment:
question: This means existing users who already have decided to share or not share their location permissions would see this prompt - is that right? Could we limit it so that only new users see this? (Or confirm with product that it is OK not to if it hasn't been confirmed already)

Reply:
I guess we could define that the location onboarding screen doesn't apply for users who have already given the app location access; that would be complicated to check in the existing framework, though.

Reply:
I have [confirmed that this is OK](https://mbta.slack.com/archives/C05QMG9GS9M/p1730221160659249?thread_ts=1730220856.035489&cid=C05QMG9GS9M).

Reply:
Thanks for checking!